crates/cxx-qt-gen/src/parser/cxxqtdata.rs

```diff
@@ -45,6 +53,69 @@ impl ParsedCxxQtData {
         Ok(())
     }
 
+    /// Search any ForeignMod's and look for a cxx_name attribute on a type
+    ///
+    /// We need to know this as it affects the type name used in the C++ generation
+    pub fn parse_cxx_names_map(&mut self, item: &Item, bridge_namespace: &str) -> Result<()> {
+        // Extract the foreign mod (extern "ABI" { ... })
+        let foreign_mod = match item {
+            Item::ForeignMod(foreign_mod) => Some(foreign_mod.clone()),
+            // Could be Verbatim TokenStream when it's an unsafe block, the remainder of the blocks are a normal ForeignMod though
+            Item::Verbatim(tokens) => verbatim_to_foreign_mod(tokens)?,
+            _others => None,
+        };
+
+        // If there is a foreign mod then process it
+        if let Some(foreign_mod) = &foreign_mod {
+            // Retrieve a namespace from the mod or the bridge
+            let block_namespace =
+                if let Some(index) = attribute_find_path(&foreign_mod.attrs, &["namespace"]) {
+                    attribute_tokens_to_value::<LitStr>(&foreign_mod.attrs[index])?.value()
+                } else {
+                    bridge_namespace.to_owned()
+                };
+
+            // Read each of the types in the mod (type A;)
+            for foreign_type in foreign_mod_to_foreign_item_types(foreign_mod)? {
+                // Retrieve the namespace for the type itself if there is one
+                let namespace =
+                    if let Some(index) = attribute_find_path(&foreign_type.attrs, &["namespace"]) {
```

Comment:
need to check if we actually need the namespace mapping. In the CXX generation it seems to add `using T = A;` inside the namespace if you have done
```rust
#[namespace = "my_namespace"]
unsafe extern "C++" {
   type T = A;
}
```

Reply:
As CXX generates this in it's header

```cpp
namespace my_namespace {
  using T = ...;
}
```

However, when you are in a method which is in another namespace the type still needs to be full qualified to reach the other namespace. So CXX in it's generation appears to always used fully qualified namespace + type in it's generation.